nmrfx-structure/src/main/java/org/nmrfx/structure/chemistry/energy/Dihedral.java

```diff
@@ -429,41 +429,24 @@ public void setBoundaries(final double sigma, boolean useDegrees, double maxRang
             Atom atom = angleAtoms.get(iAtom).daughterAtom;
             // temporarily set everything to use sigma, till we make sure counting is correct
             if (atom == null) {
-                System.out.println("daughter null " + angleAtoms.get(iAtom).daughterAtom.getFullName());
-            }
-            if (atom.parent == null) {
-                System.out.println("parent null " + angleAtoms.get(iAtom).daughterAtom.getFullName());
-            }
-            if (atom.parent.getName().equals("P") || atom.parent.getName().equals("O5'") || atom.parent.getName().equals("C5'") || atom.parent.getName().equals("C4'") || atom.parent.getName().equals("O3'")) {
-                inputSigma[aStart++] = sigma / backBoneScale;
-                if (sinCosMode) {
-                    inputSigma[aStart++] = sigma / backBoneScale;
-                }
-            } else if (atom.parent.getName().equals("CA") || atom.parent.getName().equals("C")) {
-                inputSigma[aStart++] = sigma / backBoneScale;
-                if (sinCosMode) {
-                    inputSigma[aStart++] = sigma / backBoneScale;
-                }
+                throw new NullPointerException("daughter null " + angleAtoms.get(iAtom).daughterAtom.getFullName());
+            } else if (atom.parent == null) {
+                throw new NullPointerException("parent null " + angleAtoms.get(iAtom).daughterAtom.getFullName());
```

Comment:
the if statements were specifically added as debug statements to be printed before the NPE occurred but nothing was done to stop the NPE from happening so seems like it is expected for them to be thrown here, moved the messages to the exceptions themselves. Not sure if this is okay?

Reply:
Changed to use IllegalStateException